sonar-db/src/main/java/org/sonar/db/DatabaseUtils.java

```diff
@@ -136,6 +137,19 @@ private static String escapePercentAndUnderscore(String value) {
   }
 
   /**
+   * Partition by 1000 elements a list of input and execute a consumer on each part.
+   *
+   * The goal is to prevent issue with ORACLE when there's more than 1000 elements in a 'in ('X', 'Y', ...)'
+   * and with MsSQL when there's more than 2000 parameters in a query
+   */
+  public static <INPUT> void executeLargeUpdates(Collection<INPUT> inputs, Consumer<List<INPUT>> consumer) {
+    Iterable<List<INPUT>> partitions = Iterables.partition(inputs, PARTITION_SIZE_FOR_ORACLE);
```

Comment:
please use private method `toUniqueAndSortedPartitions` it will ensure uniqness of UUIDs and sort them
using IN arguments in reproductive order is usually more DB optimizer friendly